src/renderer/components/+cluster-settings/components/cluster-metrics-setting.tsx

```diff
@@ -0,0 +1,118 @@
+import "./cluster-metrics-setting.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { Select, SelectOption } from "../../select/select";
+import { Icon } from "../../icon/icon";
+import { Button } from "../../button/button";
+import { SubTitle } from "../../layout/sub-title";
+import { Cluster } from "../../../../main/cluster";
+import { observable } from "mobx";
+
+interface Props {
+  cluster: Cluster;
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+@observer
+export class ClusterMetricsSetting extends React.Component<Props> {
+  @observable hiddenMetrics = observable.set<string>();
+
+  componentDidMount() {
+    this.hiddenMetrics = observable.set<string>(this.props.cluster.preferences.hiddenMetrics);
+  }
+
+  save = () => {
+    this.props.cluster.preferences.hiddenMetrics = Array.from(this.hiddenMetrics);
+  };
+
+  onChangeSelect(values: ResourceType[]) {
+    values.map(value => {
+      if (this.hiddenMetrics.has(value)) {
+        this.hiddenMetrics.delete(value);
+      } else {
+        this.hiddenMetrics.add(value);
+      }
+    });
+    this.save();
+  }
```

Comment:
This way we don't need the lamba in two places. Plus we really shouldn't be using `Array.map` here.

```suggestion
  onChangeSelect = (values: SelectOption<ResourceType>[]) => {
    for (const { value } of values) {
      if (this.hiddenMetrics.has(value)) {
        this.hiddenMetrics.delete(value);
      } else {
        this.hiddenMetrics.add(value);
      }
    }
    this.save();
  }
```